ovos_plugin_common_play/ocp/__init__.py

```diff
@@ -76,29 +77,32 @@ def handle_home(self):
         self.gui.show_home(app_mode=True)
 
     def register_ocp_intents(self, message=None):
-        if not self._intents_event.is_set():
-            missing = True
-        else:
-            # check list of registered intents
-            # if needed register ocp intents again
-            # this accounts for restarts etc
-            i = IntentQueryApi(self.bus)
-            intents = i.get_padatious_manifest()
-            missing = not any(e.startswith("ovos.common_play:") for e in intents)
-
-        if missing:
-            LOG.info("OCP intents missing, registering")
-            self.register_intent("play.intent", self.handle_play)
-            self.register_intent("read.intent", self.handle_read)
-            self.register_intent("open.intent", self.handle_open)
-            self.register_intent("next.intent", self.handle_next)
-            self.register_intent("prev.intent", self.handle_prev)
-            self.register_intent("pause.intent", self.handle_pause)
-            self.register_intent("resume.intent", self.handle_resume)
-            self._intents_event.set()
-
-        # trigger a presence announcement from all loaded ocp skills
-        self.bus.emit(Message("ovos.common_play.skills.get"))
+        with self._intent_registration_lock:
+            if not self._intents_event.is_set():
```

Comment:
OCP may be running standalone as a service (not via the plugin), in that case it should handle reloads of core, so this event is not safe since it if core reload the intents need reregistering